packages/wonder-blocks-core/components/provider-group.js

```diff
@@ -0,0 +1,37 @@
+// @flow
+import * as React from "react";
+
+type Provider<P> = {|
+    type: React.ComponentType<P>,
+    props?: P,
+|};
+
+type Props = {|
+    providers: Array<Provider<any>>,
+    children: (Array<any>) => React.Node,
+|};
+
+export default class ProviderGroup extends React.Component<Props> {
+    render() {
+        const {providers, children} = this.props;
+        const provided = [];
+        let i = 0;
+        const buildProviders = (...args) => {
+            if (i > 0) {
+                provided.push(...args);
+            }
+            if (i < providers.length) {
+                const providerEl = React.createElement(providers[i].type, {
+                    ...providers[i].props,
+                    children: buildProviders,
+                });
```

Comment:
For each provider, we tell it that its children is our little function. This allows us to gather its CaaF arguments into our array and then tell that provider that its child is the next provider.